### ISSUE TITLE
Unable to start spark-submit python file located in s3

### ISSUE THREAD

**Issue by MattFlower** [awslabs/emr-bootstrap-actions#113]
I'm attempting to use spark submit to start a python file located in s3.  I'm trying something like:

    /home/hadoop/spark/bin/spark-submit --master yarn-cluster --deploy-mode cluster s3://bucket/folder/folder/spark.py

I get some indication that it might be working:

  15/05/26 23:08:35 INFO yarn.Client: Uploading resource s3://bucket/folder/folder/spark.py -> hdfs://10.196.88.42:9000/user/hadoop/.sparkStaging/application_1432676762331_0011/spark.py

But in the end, the end of the script does not write to s3 as it should.  It does work correctly if I submit via a local spark.py file.  It also works correctly if I take the file and upload it to hdfs and start the job via:

    /home/hadoop/spark/bin/spark-submit --master yarn-cluster --deploy-mode cluster hdfs://10.196.88.42:9000/user/hadoop/spark.py

I can successfully reach the s3 file from my cluster using "aws s3 cp s3://bucket/folder/folder/spark.py .", so the role permissions seem to be correct.

Although you documentation indicates this works with jar files, I suspect it may be broken for pyspark.

**Comment by christopherbozeman** [comment]
This was more of a restriction on spark-submit itself with prior versions, but I believe this changed with Spark 1.3. For example, I uploaded the built-in pi.py example ( ~/spark/examples/src/main/python/pi.py) to S3 and was able to able execute successfully directly at CLI with spark-submit using ` --master yarn --deploy-mode cluster`  as well as through EMR Step using script-runner.  I tested this with EMR AMI 3.7.0 which currently installs Spark 1.3.1 by default.
